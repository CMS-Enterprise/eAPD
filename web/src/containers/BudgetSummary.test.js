import { shallow } from 'enzyme';
import React from 'react';

import {
  plain as BudgetSummary,
  mapStateToProps,
  DataRow,
  DataRowDetails,
  DataRowGroup,
  HeaderRow
} from './BudgetSummary';

describe('budget summary component', () => {
  test('renders correctly', () => {
    const component = shallow(
      <BudgetSummary
        activities={{
          hie: ['hie data'],
          hit: ['hit data'],
          mmis: ['mmis data']
        }}
        data={{
          hie: {},
          hit: {},
          mmis: {},
          combined: {
            '1': { federal: 1, state: 2, medicaid: 1000, total: 3 },
            '2': { federal: 10, state: 20, medicaid: 2000, total: 30 }
          }
        }}
        years={['1', '2']}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test('data row renders with no data', () => {
    expect(
      shallow(
        <DataRow
          category="category text"
          data={{
            total: { total: 0 }
          }}
          entries={[]}
          title="title text"
        />
      )
    ).toMatchSnapshot();
  });

  test('data row renders with data', () => {
    const component = shallow(
      <DataRow
        category="category text"
        data={{
          '1': { federal: 1, state: 2, medicaid: 1000, total: 3 },
          '2': { federal: 10, state: 20, medicaid: 2000, total: 30 },
          total: { federal: 100, state: 200, medicaid: 10000, total: 300 }
        }}
        entries={[]}
        title="title text"
      />
    );
    expect(component).toMatchSnapshot();

    // Toggle details
    component.find('button').simulate('click');
    expect(component).toMatchSnapshot();
  });

  test('data row details renders correctly', () => {
    expect(
      shallow(
        <DataRowDetails
          category="category"
          entries={[
            {
              name: 'activity 1',
              id: 1,
              data: { category: { '1': 1, '2': 2, total: 3 } }
            },
            {
              id: 2,
              data: { category: { '1': 10, '2': 20, total: 30 } }
            }
          ]}
          years={['1', '2', 'total']}
        />
      )
    ).toMatchSnapshot();
  });

  test('data row group renders', () => {
    expect(
      shallow(
        <DataRowGroup
          data={{
            combined: {},
            contractors: {},
            expenses: {},
            statePersonnel: {}
          }}
          entries={[10, 20, 30]}
        />
      )
    ).toMatchSnapshot();
  });

  test('header row renders', () => {
    expect(
      shallow(<HeaderRow title="row title" years={['1', '2']} />)
    ).toMatchSnapshot();
  });

  test('maps state to props', () => {
    const state = {
      apd: { data: { years: ['1', '2'] } },
      budget: {
        activityTotals: [
          { fundingSource: 'HIE', name: 'activity 1' },
          { fundingSource: 'HIT', name: 'activity 2' },
          { fundingSource: 'MMIS', name: 'activity 3' },
          { fundingSource: 'MMIS', name: 'activity 4' }
        ]
      }
    };

    expect(mapStateToProps(state)).toEqual({
      activities: {
        hie: [{ fundingSource: 'HIE', name: 'activity 1' }],
        hit: [{ fundingSource: 'HIT', name: 'activity 2' }],
        mmis: [
          { fundingSource: 'MMIS', name: 'activity 3' },
          { fundingSource: 'MMIS', name: 'activity 4' }
        ]
      },
      data: {
        activityTotals: [
          { fundingSource: 'HIE', name: 'activity 1' },
          { fundingSource: 'HIT', name: 'activity 2' },
          { fundingSource: 'MMIS', name: 'activity 3' },
          { fundingSource: 'MMIS', name: 'activity 4' }
        ]
      },
      years: ['1', '2']
    });
  });
});
