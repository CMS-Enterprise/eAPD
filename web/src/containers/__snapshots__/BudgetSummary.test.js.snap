// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`budget summary component data row group renders 1`] = `
<Fragment>
  <DataRow
    category="combined"
    data={Object {}}
    entries={
      Array [
        10,
        20,
        30,
      ]
    }
    key="combined"
    title="Subtotal"
    year="1448"
  />
  <DataRow
    category="contractors"
    data={Object {}}
    entries={
      Array [
        10,
        20,
        30,
      ]
    }
    key="contractors"
    title="Contracted Resources"
    year="1448"
  />
  <DataRow
    category="expenses"
    data={Object {}}
    entries={
      Array [
        10,
        20,
        30,
      ]
    }
    key="expenses"
    title="Non-Personnel"
    year="1448"
  />
  <DataRow
    category="statePersonnel"
    data={Object {}}
    entries={
      Array [
        10,
        20,
        30,
      ]
    }
    key="statePersonnel"
    title="Project State Staff"
    year="1448"
  />
</Fragment>
`;

exports[`budget summary component data row renders with data 1`] = `
<tr
  className=""
>
  <th
    scope="row"
  >
    title text
  </th>
  <td
    className="budget-table--number"
  >
    <Dollars
      long={false}
    >
      1000
    </Dollars>
  </td>
  <td
    className="budget-table--number"
  >
    <Dollars
      long={false}
    >
      1
    </Dollars>
  </td>
  <td
    className="budget-table--number"
  >
    <Dollars
      long={false}
    >
      2
    </Dollars>
  </td>
</tr>
`;

exports[`budget summary component data row renders with no data 1`] = `
<tr
  className=""
>
  <th
    scope="row"
  >
    title text
  </th>
  <td
    className="budget-table--number"
  >
    <Dollars
      long={false}
    >
      0
    </Dollars>
  </td>
  <td
    className="budget-table--number"
  >
    <Dollars
      long={false}
    >
      0
    </Dollars>
  </td>
  <td
    className="budget-table--number"
  >
    <Dollars
      long={false}
    >
      0
    </Dollars>
  </td>
</tr>
`;

exports[`budget summary component header row renders 1`] = `
<tr>
  <th
    id="summary-budget-fy-1"
    key="1"
  >
    FFY 
    1
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    Medicaid Total
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    Federal Total
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    State Total
  </th>
</tr>
`;

exports[`budget summary component renders correctly 1`] = `
<Fragment>
  <h3
    className="ds-h3"
  >
    HIT Activities
  </h3>
  <table
    className="budget-table"
    key="1"
  >
    <thead>
      <HeaderRow
        yr="1"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "hit data",
          ]
        }
        year="1"
      />
    </tbody>
  </table>
  <table
    className="budget-table"
    key="2"
  >
    <thead>
      <HeaderRow
        yr="2"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "hit data",
          ]
        }
        year="2"
      />
    </tbody>
  </table>
  <table
    className="budget-table"
    key="total"
  >
    <thead>
      <HeaderRow
        yr="total"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "hit data",
          ]
        }
        year="total"
      />
    </tbody>
  </table>
  <h3
    className="ds-h3"
  >
    HIE Activities
  </h3>
  <table
    className="budget-table"
    key="1"
  >
    <thead>
      <HeaderRow
        yr="1"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "hie data",
          ]
        }
        year="1"
      />
    </tbody>
  </table>
  <table
    className="budget-table"
    key="2"
  >
    <thead>
      <HeaderRow
        yr="2"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "hie data",
          ]
        }
        year="2"
      />
    </tbody>
  </table>
  <table
    className="budget-table"
    key="total"
  >
    <thead>
      <HeaderRow
        yr="total"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "hie data",
          ]
        }
        year="total"
      />
    </tbody>
  </table>
  <h3
    className="ds-h3"
  >
    MMIS Activities
  </h3>
  <table
    className="budget-table"
    key="1"
  >
    <thead>
      <HeaderRow
        yr="1"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "mmis data",
          ]
        }
        year="1"
      />
    </tbody>
  </table>
  <table
    className="budget-table"
    key="2"
  >
    <thead>
      <HeaderRow
        yr="2"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "mmis data",
          ]
        }
        year="2"
      />
    </tbody>
  </table>
  <table
    className="budget-table"
    key="total"
  >
    <thead>
      <HeaderRow
        yr="total"
      />
    </thead>
    <tbody>
      <DataRowGroup
        data={Object {}}
        entries={
          Array [
            "mmis data",
          ]
        }
        year="total"
      />
    </tbody>
  </table>
  <h3
    className="ds-h3"
  >
    Project Activities Totals
  </h3>
  <table
    className="budget-table"
  >
    <thead>
      <tr>
        <th
          id="summary-budget-null1"
        />
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          Medicaid Total
        </th>
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          Federal Total
        </th>
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          State Total
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="1"
      >
        <th
          scope="row"
        >
          FFY 
          1
        </th>
        <td
          className="budget-table--number"
        >
          <Dollars
            long={false}
          >
            1000
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars
            long={false}
          >
            1
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars
            long={false}
          >
            2
          </Dollars>
        </td>
      </tr>
      <tr
        key="2"
      >
        <th
          scope="row"
        >
          FFY 
          2
        </th>
        <td
          className="budget-table--number"
        >
          <Dollars
            long={false}
          >
            2000
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars
            long={false}
          >
            10
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars
            long={false}
          >
            20
          </Dollars>
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
